Revamp Logging documentations in 0.19 (#2665)
kedro-org/kedro: docs/source/logging/index.md

To add logging to your own code (e.g. in a node):

```python
import logging

logger = logging.getLogger(__name__)
logger.warning("Issue warning")
logger.info("Send information")
logger.debug("Useful information for debugging")
```

You can use Rich's [console markup](https://rich.readthedocs.io/en/stable/markup.html) in your logging calls:

```python
log.error("[bold red blink]Important error message![/]", extra={"markup": True})
```

## How to customise Kedro logging

To customise logging in your Kedro project, you need to specify the path to a project-specific logging configuration file. Change the environment variable `KEDRO_LOGGING_CONFIG` to override the default logging configuration. Point the variable instead to your project-specific configuration, which we recommend you store inside the project's`conf` folder, and name `logging.yml`.

For example, you can set `KEDRO_LOGGING_CONFIG` by typing the following into your terminal:

```bash
export KEDRO_LOGGING_CONFIG=<project_root>/conf/logging.yml
```

After setting the environment variable, any subsequent Kedro commands use the logging configuration file at the specified path.

```{note}

> If the `KEDRO_LOGGING_CONFIG` environment variable is not set, Kedro will default to using the logging configuration file at the project's default location of Kedro's `default_logging.yml`.

Reviewer: This doesn't quite make sense but I don't know how to fix it @noklam 

I think maybe it's

```suggestion
If the `KEDRO_LOGGING_CONFIG` environment variable is not set, Kedro uses the logging configuration file at the project's default location, which is named`default_logging.yml`.
```

Where is this default location?

```

### How to show DEBUG level messages
To see `DEBUG` level messages, change the level of logging in your project-specific logging configuration file (`logging.yml`). We provide a `logging.yml` template:

<details>
<summary><b>Click to expand the <code>logging.yml</code> template</b></summary>
<code>

```yaml
version: 1

disable_existing_loggers: False

formatters:
  simple:
    format: "%(asctime)s - %(name)s - %(levelname)s - %(message)s"

handlers:
  console:
    class: logging.StreamHandler
    level: INFO
    formatter: simple
    stream: ext://sys.stdout

  info_file_handler:
    class: logging.handlers.RotatingFileHandler
    level: INFO
    formatter: simple
    filename: info.log
    maxBytes: 10485760 # 10MB
    backupCount: 20
    encoding: utf8
    delay: True

  rich:
    class: kedro.logging.RichHandler
    rich_tracebacks: True
    # Advance options for customisation.
    # See https://docs.kedro.org/en/stable/logging/logging.html#project-side-logging-configuration
    # tracebacks_show_locals: False

loggers:
  kedro:
    level: INFO

  your_python_package:
    level: INFO

root:
  handlers: [rich]
```

> </code>

Reviewer 2: Not sure the code block should end here, the docs end up rendering as follows: 
![image](https://github.com/kedro-org/kedro/assets/90615669/30bfec0c-bbbd-47bc-ae54-02df1fa16b1e)

Author: ![image](https://github.com/kedro-org/kedro/assets/18221871/b940869c-9de1-4e76-a79e-c3c1233e54a2)
 Before you expand the template it looks pretty normal. But I agree it looks weird after it get expanded. I will add a short line there.

# Logging

> Kedro uses [Python's `logging` library](https://docs.python.org/3/library/logging.html). Configuration is provided as a dictionary according to the [Python logging configuration schema](https://docs.python.org/3/library/logging.config.html#logging-config-dictschema) in [Default configuration built into the Kedro framework](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml).

Reviewer 2: ```suggestion
Kedro uses [Python's `logging` library](https://docs.python.org/3/library/logging.html). Configuration is provided as a dictionary according to the [Python logging configuration schema](https://docs.python.org/3/library/logging.config.html#logging-config-dictschema) in [Kedro's default logging configuration](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml).
```

> You can alter this configuration and provide different configurations for different run environment according to the [standard Kedro mechanism for handling configuration](../configuration/configuration_basics.md).

Reviewer 3: This isn't true any more. The only way to alter the configuration is through the environment variable, and you can't change it to be different in different run environments.

We should be able to simplify this way more. I think the note and everything that follows it could just be deleted?

Ultimately maybe we don't even need more than one page on logging since there won't be much to say. But I'm not sure what the current scheme is for what goes in index.md compared to other files.
